src/agent/src/image_rpc.rs

```diff
@@ -211,7 +120,7 @@ impl ImageService {
         let cid = if !req_cid.is_empty() {
             req_cid.to_string()
         } else if let Some(last) = req.get_image().rsplit('/').next() {
-            // ':' have special meaning for umoci during upack
+            // ':' not valid for container id
             last.replace(':', "_")
```

Comment:
After remove umoci, we don't need this replace operate. If cid is empty, we will use this info as image name which can associate with tags or digest

Reply:
Hey Arron, do you mean replacing `last.replace(':', "_")` with just `last.to_string()` in this `else if`? I initially did this and updates the two unit tests that check the replacement of `:` with `_`, but we got an error from validate.rs (https://github.com/kata-containers/kata-containers/blob/40c69043245d0326080f48b2d295d31eee52e385/src/libs/kata-sys-util/src/validate.rs#L7-L28) 
`invalid container ID: "a:b"`
That's why I added the replace back in, but maybe there is a better way to solve this?

Reply:
> Hey Arron, do you mean replacing `last.replace(':', "_")` with just `last.to_string()` in this `else if`? I initially did this and updates the two unit tests that check the replacement of `:` with `_`, but we got an error from validate.rs (
> 
> https://github.com/kata-containers/kata-containers/blob/40c69043245d0326080f48b2d295d31eee52e385/src/libs/kata-sys-util/src/validate.rs#L7-L28
> 
> )
> `invalid container ID: "a:b"`
> That's why I added the replace back in, but maybe there is a better way to solve this?

Hi Steven, my fault, this commit `cde438ceb781888a8e6eadd9d31033efb0a3628c` "image_rpc: always call `verify_cid` after determining the cid" added cid check when the cid was derived from the image path. The logic here is correct.